File: backend/internal/model/masking_rule.go
package model

import (
	"encoding/json"
	"time"
)

type MaskingRule struct {
	ID         uint64          `db:"id" json:"id"`
	ColumnName string          `db:"column_name" json:"column_name"`
	MatchType  string          `db:"match_type" json:"match_type"`
	MaskMode   string          `db:"mask_mode" json:"mask_mode"`
	MaskConfig json.RawMessage `db:"mask_config" json:"mask_config"`
	Enabled    bool            `db:"enabled" json:"enabled"`
	CreatedBy  uint64          `db:"created_by" json:"created_by"`
	CreatedAt  time.Time       `db:"created_at" json:"created_at"`
}

File: frontend/src/shared/types/maskingRule.ts
export type MaskingRule = {
  id: number
  column_name: string
  match_type: 'exact' | 'regex'
  mask_mode: 'full' | 'partial' | 'hash' | 'email' | 'fixed' | 'numeric' | 'datetime' | 'ip'
  mask_config: Record<string, unknown> | null
  created_by: number
  created_at: string
}

export type MaskingWhitelist = {
  id: number
  db_connection_id: number
  database_name: string
  schema_name?: string
  table_name: string
  column_name: string
  created_by: number
  created_at: string
}

export type RedisSensitiveKeyPrefix = {
  id: number
  db_connection_id: number
  redis_db_index?: number | null
  key_prefix: string
  reason?: string | null
  is_active: boolean
  created_by?: number | null
  created_at: string
  updated_at: string
}
